core/src/operations/decorators/advice.rs

```diff
@@ -274,13 +274,18 @@ pub enum AdviceInjector {
     ///
     /// Outputs:
     ///   Operand stack: [PK, MSG, ...]
-    ///   Advice stack: [SIG_DATA]
+    ///   Advice stack: \[SIG_DATA\]
     ///
     /// Where PK is the public key corresponding to the signing key, MSG is the message, SIG_DATA
     /// is the signature data.
     SigToStack { kind: SignatureKind },
 }
-
+#[cfg(feature = "formatter")]
+impl crate::prettier::PrettyPrint for AdviceInjector {
+    fn render(&self) -> crate::prettier::Document {
+        crate::prettier::display(self)
+    }
+}
```

Comment:
I'll see if there is a rustfmt rule for this and enable it if so, since I rely on rustfmt to handle things like this across different codebases. In any case, I often collapse trivial trait implementations to keep as much code on screen as possible, but I'm totally fine with extra whitespace for this, just want to make sure the formatter is handling it for me in case I fall back into my default habit.

Reply:
It looks like the closest thing would be `blank_lines_lower_bound`, but it appears to be totally broken at the moment (and is unstable to boot), so I guess this will have to be manually enforced for the time being.

However, I also noticed that we have a style guideline about imports, but we've disabled the relevant rustfmt rules that would enforce it, namely:

```
group_imports = "StdExternalCrate"
imports_granularity = "Crate"
```

We should definitely enable those if we actually want to enforce that style consistently everywhere. I've already noti ced many places where the style around imports differs (though I attempted to resolve those/be consistent in all of the code I modified/introduced).

Reply:
Yes - we do this in other repos (e.g., [here](https://github.com/0xPolygonMiden/miden-base/blob/main/rustfmt.toml)), but haven't gotten to doing it in this repo yet.